src/UseCases/ApplyForMembership/ApplyForMembershipUseCase.php

```diff
@@ -93,12 +89,22 @@ public function applyForMembership( ApplyForMembershipRequest $request ): ApplyF
 		return ApplyForMembershipResponse::newSuccessResponse(
 			$tokens->getAccessToken(),
 			$tokens->getUpdateToken(),
-			$application
+			$application,
+			$paymentCreationResponse->paymentProviderURLGenerator->generateURL( $requestContext )
 		);
 	}
 
-	private function newApplicationFromRequest( ApplyForMembershipRequest $request ): MembershipApplication {
-		return ( new MembershipApplicationBuilder( $this->incentiveFinder ) )->newApplicationFromRequest( $request );
+	private function newApplicationFromRequest( ApplyForMembershipRequest $request, int $paymentId ): MembershipApplication {
+		return ( new MembershipApplicationBuilder( $this->incentiveFinder ) )->newApplicationFromRequest( $request, $paymentId );
+	}
+
+	private function extractPaymentRequestDataForMembershipApplication( ApplyForMembershipRequest $request ): PaymentCreationRequest {
+		$paymentRequest = new PaymentCreationRequest(
+			$request->getPaymentAmountInEuros()->getEuroCents(),
+			$request->getPaymentIntervalInMonths(),
+			$request->getPaymentType(),
+			// TODO get bank data (iban, bic, transfercode)?
```

Comment:
The payment creation request should be done outside the use case (by the controller)